core/src/test/scala/org/apache/spark/partial/SumEvaluatorSuite.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.partial
+
+
+import org.apache.spark._
+import org.apache.spark.util.StatCounter
+
+class SumEvaluatorSuite extends SparkFunSuite with SharedSparkContext {
+
+  test("correct handling of count 1") {
+
+    //setup
+    val counter = new StatCounter(List(2.0))
+    // count of 10 because it's larger than 1,
+    // and 0.95 because that's the default
+    val evaluator = new SumEvaluator(10, 0.95)
+    // arbitrarily assign id 1
+    evaluator.merge(1, counter)
+
+    //execute
+    val res = evaluator.currentResult()
+    // 38.0 - 7.1E-15 because that's how the maths shakes out
+    val target_mean =  38.0 - 7.1E-15
```

Comment:
My python is showing 😬

On Thursday, March 31, 2016, Sean Owen <notifications@github.com> wrote:

> In core/src/test/scala/org/apache/spark/partial/SumEvaluatorSuite.scala
> <https://github.com/apache/spark/pull/12016#discussion_r58143122>:
>
> > +class SumEvaluatorSuite extends SparkFunSuite with SharedSparkContext {
> > +
> > +  test("correct handling of count 1") {
> > +
> > +    //setup
> > +    val counter = new StatCounter(List(2.0))
> > +    // count of 10 because it's larger than 1,
> > +    // and 0.95 because that's the default
> > +    val evaluator = new SumEvaluator(10, 0.95)
> > +    // arbitrarily assign id 1
> > +    evaluator.merge(1, counter)
> > +
> > +    //execute
> > +    val res = evaluator.currentResult()
> > +    // 38.0 - 7.1E-15 because that's how the maths shakes out
> > +    val target_mean =  38.0 - 7.1E-15
>
> Looking good, now into nits. var names should be camelCase instead of
> with_underscores.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12016/files/93bcd99b11e0c914053cb1cfbe37fe0df4ba1a38#r58143122>
>

-- 
Want to work at Handy? Check out our culture deck and open roles 
<http://www.handy.com/careers>
Latest news <http://www.handy.com/press> at Handy
Handy just raised $50m 
<http://venturebeat.com/2015/11/02/on-demand-home-service-handy-raises-50m-in-round-led-by-fidelity/> led 
by Fidelity